test/e2e/far_e2e_test.go

```diff
@@ -39,12 +69,65 @@ var _ = Describe("FAR E2e", func() {
 			Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(far), testFarCR)).To(Succeed(), "failed to get FAR CR")
 		})
 	})
+
+	Context("fence agent - non-Dummy", func() {
+		BeforeEach(func() {
+			var testNodeName string
+			nodes := &corev1.NodeList{}
+
+			Expect(k8sClient.List(context.Background(), nodes, &client.ListOptions{})).ToNot(HaveOccurred())
+			if len(nodes.Items) <= 1 {
+				Fail("there is one or less available nodes in the cluster")
+			}
+			//TODO: Randomize the node selection
+			// run FA on the first node - a master node
+			nodeObj := nodes.Items[nodeIndex]
+			testNodeName = nodeObj.Name
+			log.Info("Testing Node", "Node name", testNodeName)
+
+			if clusterPlatformType == "AWS" {
+				fenceAgent = fenceAgentAWS
+				By("running fence_aws")
+			} else if clusterPlatformType == "BareMetal" {
+				fenceAgent = fenceAgentIPMI
+				By("running fence_ipmilan")
+			} else {
+				Skip("FAR haven't been tested on this kind of cluster (non AWS or BareMetal)")
+			}
+
+			testShareParam, err := buildSharedParameters(clusterPlatform, fenceAgentAction)
+			if err != nil {
+				Fail("can't get shared information")
+			}
+			testNodeParam, err := buildNodeParameters(clusterPlatformType)
+			if err != nil {
+				Fail("can't get node information")
+			}
+
+			far = createFAR(testNodeName, fenceAgent, testShareParam, testNodeParam)
+		})
+
+		AfterEach(func() {
+			deleteFAR(far)
```

Comment:
I can't since FAR is still using Ginkgo v1. Once this PR is merged and we have good enough E2E test then I will update tools versions. In the process I will try to update it with some cool changes in the new versions, such as `DeferCleanup`.